cosmo_tester/test_suites/cluster/full_cluster_test.py

```diff
@@ -378,6 +382,94 @@ def test_three_nodes_cluster_status(three_nodes_cluster, logger):
                                         node1.client)
 
 
+def test_three_nodes_cluster_teardown(three_nodes_cluster, ssh_key,
+                                      test_config, module_tmpdir, logger):
+    """Tests a cluster teardown"""
+    node1, node2, node3 = three_nodes_cluster
+    nodes_list = [node1, node2, node3]
+    logger.info('Asserting cluster status')
+    _assert_cluster_status(node1.client)
+
+    logger.info('Installing example deployment')
+    example = get_example_deployment(node1, ssh_key, logger,
+                                     'cluster_teardown', test_config)
+    example.inputs['server_ip'] = node1.ip_address
+    example.upload_and_verify_install()
+
+    logger.info('Removing example deployment')
+    example.uninstall()
+    logger.info('Removing cluster')
+    for node in nodes_list:
+        for config_name in ['db', 'rabbit', 'manager']:
+            node.run_command('cfy_manager remove -v -c /etc/cloudify/'
+                             '{0}_config.yaml'.format(config_name))
+
+    credentials = _get_new_credentials()
+    logger.info('New credentials: %s', credentials)
+
+    for node in nodes_list:
+        node.install_config = copy.deepcopy(node.basic_install_config)
+
+    logger.info('Installing Cloudify cluster again')
+    run_cluster_bootstrap(nodes_list, nodes_list, nodes_list,
+                          skip_bootstrap_list=[], pre_cluster_rabbit=True,
+                          high_security=True, use_hostnames=False,
+                          tempdir=module_tmpdir, test_config=test_config,
+                          logger=logger, revert_install_config=True,
```

Comment:
Maybe I'm misreading something, but it looks like you revert the install config /after/ this second bootstrap, which I think won't do anything- and you're reverting it to the basic (single node) install config, which wouldn't work if you did use it.

Reply:
I might be misunderstanding something, but each node's `install_config` is reverted before the second installation. Then, during the cluster bootstrap, after each service installation on each node, the `install_config` is reverted based on the explanation above.

Reply:
Right, makes sense with your description above- I'm happyish with this for the moment, though the test is a little hard to follow but I think a lot of that is because the process itself is quite complex so I'll green it.